# 1 Assignment

It is totally prohibited to use any kind of loop. You can use stackoverflow. If you copy codes from previous answers, explain each step. No explanation is `0 points`.

## 1.1 Tuple



1. Print the first item of the second item of `tuple1` object. **Hint: Use indexing**<br><br>
2. Print the last item of the `tuple1` object.**Hint: Use indexing**<br><br>
3. Change the value "orange" for "pink grapefruit". Comment the output. **Hint: Is it possible?** <br><br>
4. Print the length of `tuple1`. **Hint: Length function**<br><br>
5. Sum all the elements of tuple2 and describe your steps and explain each of them. **Hint: Use `map` function.**<br><br>
6. Generate a new tuple object named as `tuple3` with the half values of `tuple2`. This tuple should be similar as `tuple2`, tuple of lists.  **Hint: Use `map` funciont.**<br><br>

In [22]:
tuple1 = ("Orange", [10, 20, 30], (5, 15, 25))
tuple2 = ([7, 8], [9, 1], [10, 7])

# 1.1.1 Print the first item of the second item of tuple1 object. Hint: Use indexing

In [4]:
tuple1 = ("Orange", [10, 20, 30], (5, 15, 25))
tuple1

('Orange', [10, 20, 30], (5, 15, 25))

The first thing we have to do is create the "tuple1" object and run it to see its elements

In [16]:
print(tuple1[1][0])

10


To execute print, we locate the second item of the object "tuple1": a list "[10, 20, 30]" and reference it with [1]. Then, we locate the first item in this list which is 10 and reference it with [0]. Finally it remains: print(object that is tuple1, [1] which is the second item and [0] which is the first item of the second item

# 1.1.2. Print the last item of the tuple1 object. Hint: Use indexing

In [17]:
tuple1 = ("Orange", [10, 20, 30], (5, 15, 25))
tuple1

('Orange', [10, 20, 30], (5, 15, 25))

The first thing we have to do is create the "tuple1" object and run it to see its elements

To run print on the last element, there are two options: First, we can count the elements of tuple 1: "Orange", [10, 20, 30] and (5, 15, 25) and determine that the last item is in the order [2]. So the code would look like this:

In [19]:
print(tuple1[2])

(5, 15, 25)


The second way uses negative indexing. For this, [-1] is used to find the last item of an object. In this case tuple1[-1] shows the last item of tuple1

In [20]:
print(tuple1[-1])

(5, 15, 25)


# 1.1.3. Change the value "orange" for "pink grapefruit". Comment the output. Hint: Is it possible?

"Tuples" as objects are immutable. That is to say, its elements cannot be modified once they have been declared. However, we can try to recreate a tuple using the elements of tuple1

In [23]:
tuple1 = ("pink grapefruit", tuple1[1], tuple1[2])
print(tuple1)

('pink grapefruit', [10, 20, 30], (5, 15, 25))


# 1.1.4. Print the length of tuple1. Hint: Length function

In [24]:
tuple1 = ("Orange", [10, 20, 30], (5, 15, 25))
tuple1

('Orange', [10, 20, 30], (5, 15, 25))

The first thing we have to do is create the "tuple1" object and run it to see its elements

In [25]:
print(len(tuple1))

3


Print is used to know the result value of the length (len) function applied to the tuple1 object.

# 1.1.5. Sum all the elements of tuple2 and describe your steps and explain each of them. Hint: Use map function.

In [27]:
tuple2 = ([7, 8], [9, 1], [10, 7])
tuple2

([7, 8], [9, 1], [10, 7])

The first thing we have to do is create the "tuple2" object and run it to see its elements. The elementos are 3 list: [7, 8] , [9, 1] y [10, 7]

In [42]:
suma_elementos = sum(map(sum, tuple2))
suma_elementos

42

To add up the elements of the tuple2 object we use the "sum" function, but this function cannot add lists, so we must first add up the elements of each list and then add the results of each list.
For this, the "map" function is used, which applies the "sum" function to all the elements of tuple2 "map(sum, tuple2)". Then, the results of the "map" function are added using the "sum" function: "sum(map(sum, tuple2)"

# 1.1.6 Generate a new tuple object named as tuple3 with the half values of tuple2. This tuple should be similar as tuple2, tuple of lists. Hint: Use map funciont

In [46]:
tuple2 = ([7, 8], [9, 1], [10, 7])
tuple2

([7, 8], [9, 1], [10, 7])

The first thing we have to do is create the "tuple2" object and run it to see its elements.

In [47]:
tuple3 = tuple(map(lambda sublist: [x / 2 for x in sublist], tuple2))
print(tuple3)

([3.5, 4.0], [4.5, 0.5], [5.0, 3.5])


First, each element in the lists of tuple2 is divided into 2. For this, the lambda expression is used to divide by 2 (x/2 : 2):

lambda sublist: [x / 2 for x in sublist]

Then, the map function is applied to cover all values divided by 2 (lambda expression) of tuple 2:

map(lambda sublist: [x / 2 for x in sublist], tuple2)

Finally, the tuple3 object is declared taking the elements from the result of the map function:

tuple3 = tuple(map(lambda sublist: [x / 2 for x in sublist], tuple2))

Print of tuple 3 is executed, printing the result: ([3.5, 4.0], [4.5, 0.5], [5.0, 3.5])
